client/components/Windows/SearchResults.vue

```diff
@@ -0,0 +1,192 @@
+<template>
+	<div id="chat-container" class="window">
+		<div
+			id="chat"
+			:class="{
+				'colored-nicks': $store.state.settings.coloredNicks,
+				'show-seconds': $store.state.settings.showSeconds,
+			}"
+		>
+			<div
+				class="chat-view"
+				data-type="search-results"
+				aria-label="Search results"
+				role="tabpanel"
+			>
+				<div class="header">
+					<SidebarToggle />
+					<span class="title"
+						>Search results for "{{ $route.params.term }}" in
+						{{ $route.params.target }}</span
+					>
+					<span class="topic"></span>
+					<MessageSearchForm :network="network" :channel="channel" />
+				</div>
+				<div class="chat-content">
+					<div ref="chat" class="chat" tabindex="-1">
+						<div v-show="moreResultsAvailable" class="show-more">
+							<button
+								ref="loadMoreButton"
+								:disabled="
+									$store.state.messageSearchInProgress ||
+										!$store.state.isConnected
+								"
+								class="btn"
+								@click="onShowMoreClick"
+							>
+								<span v-if="$store.state.messageSearchInProgress">Loading…</span>
+								<span v-else>Show older messages</span>
+							</button>
+						</div>
+
+						<div v-if="$store.state.messageSearchInProgress" class="search-status">
+							Searching...
+						</div>
+						<div v-else-if="!messages.length" class="search-status">
+							No results found.
+						</div>
+						<div
+							v-else
+							class="messages"
+							role="log"
+							aria-live="polite"
+							aria-relevant="additions"
+						>
+							<template v-for="(message, id) in messages">
+								<DateMarker
+									v-if="shouldDisplayDateMarker(message, id)"
+									:key="message.time + '-date'"
+									:message="message"
+								/>
+								<Message
+									:key="message.time"
+									:channel="channel"
+									:network="network"
+									:message="message"
+								/>
+							</template>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import socket from "../../js/socket";
+
+import SidebarToggle from "../SidebarToggle.vue";
+import Message from "../Message.vue";
+import MessageSearchForm from "../MessageSearchForm.vue";
+import DateMarker from "../DateMarker.vue";
+
+export default {
+	name: "SearchResults",
+	components: {
+		SidebarToggle,
+		Message,
+		DateMarker,
+		MessageSearchForm,
+	},
+	data() {
+		return {
+			offset: 0,
+			moreResultsAvailable: false,
+		};
+	},
+	computed: {
+		search() {
+			return this.$store.state.messageSearchResults;
+		},
+		messages() {
+			if (!this.search) {
+				return [];
+			}
+
+			return this.search.results.slice().reverse();
```

Comment:
Is this sorting so that recent messages are at the bottom?

I think search should operate in the way that recent messages are at the top, and going down they're older. So the load button should be moved to the bottom, and new results are just appended.

Reply:
I don't have a strong opinion on this. I did the "newest at the bottom" order since that's what we do in everywhere else. The opposite order will probably require some tweaking of the date markers and such (if we decide to keep them),